File: CineTrack.api/Dtos/UserAnalyticsDto.cs
namespace CineTrack.api.Dtos;

public class UserAnalyticsDto
{
    public int TotalMovies { get; set; }
    public int  WatchedMovies { get; set; }
    public int TotalWatchTimeMinutes { get; set; }
    public List<GenreStatDto> FavouriteGenres { get; set; } = new List<GenreStatDto>();
    public List<MonthlyStatDto> MonthlyStats { get; set; } = new List<MonthlyStatDto>();
}

File: CineTrack.front/src/app/features/analytics/models/user-analytics-dto.ts
import { FavouriteGenre } from './FavouriteGenre';
import { MonthlyStat } from './MonthlyStat';

export interface UserAnalyticsDto {
  totalMovies: number;
  watchedMovies: number;
  totalWatchTimeMinutes: number;
  favouriteGenres: FavouriteGenre[];
  monthlyStats: MonthlyStat[];
}
